commerce_coordinator/apps/iap/api/v1/views.py

```diff
@@ -0,0 +1,188 @@
+from commerce_coordinator.apps.iap.api.v1.serializer import CreateOrderSerializer, OrderResponseSerializer
+
+logger = logging.getLogger(__name__)
+
+PAYPAL_PAYMENT_SERVICE_PROVIDER = "PayPal"
+
+class CreateOrderView(APIView):
+    """
+    API view for creating an order from a commercetools cart and payment.
+    Handles validation, optional shipping, PayPal payment creation,
+    payment attachment, and order creation with state transitions.
+    """
+    permission_classes = (IsAuthenticated,)
+
+    # pylint: disable=too-many-statements
+    def post(self, request):
+        """Handles POST request to create an order from a cart and payment."""
+        serializer = CreateOrderSerializer(data=request.data)
+        if not serializer.is_valid():
+            logger.warning(f"Order creation failed due to invalid input: {serializer.errors}")
+            return JsonResponse({'error': serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        cart_id = data["cart_id"]
+        order_number = data["order_number"]
+        payment_method = data["payment_method"]
+        shipping_address_data = data.get("shipping_address")
+        paypal_order = data.get("paypal_order") or {
+            "id": "TEST12345",
+            "status": "APPROVED",
+            "paymentSource": {"type": payment_method or "paypal"}
+        }
+
+        try:
+            ct_client = CommercetoolsAPIClient()
+
+            cart = ct_client.base_client.carts.get_by_id(cart_id)
+
+            if not cart.customer_id or str(cart.customer_id) != str(request.user.customer_id):
+                logger.warning(
+                    f"[CreateOrderView] Unauthorized cart access attempt by user "
+                    f"[{request.user.id}] on cart [{cart.id}]."
+                )
+                return JsonResponse({'error': 'Unauthorized access to cart.'}, status=status.HTTP_403_FORBIDDEN)
+
+            if not is_cart_active(cart):
+                logger.error(f"[CreateOrderView] Cart {cart_id} is not active. Current state: {cart.cart_state}")
+                return Response(
+                    {"error": f"Cart {cart_id} is not in active state. Current state: {cart.cart_state}"},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+
+            try:
+                cart = set_shipping_address(cart, shipping_address_data, ct_client)
+            except ValueError as e:
+                logger.error(f"[CreateOrderView] ValueError encountered: {str(e)}")
+                return Response({"error": "Invalid shipping address"}, status=status.HTTP_400_BAD_REQUEST)
+            except CommercetoolsError as e:
+                logger.error(f"[CreateOrderView] Error setting shipping address: {str(e)}")
+                return Response({"error": "Failed to process shipping address."},
+                                status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
i think status should be BAD Request and not server error. what is the behaviour of legacy ecommerce. Also, we should retry in case of CommercetoolsError